python/cudf/cudf/tests/test_dataframe_copy.py

```diff
@@ -161,7 +163,10 @@ def test_kernel_deep_copy():
     sr = gdf["b"]
 
     add_one[1, len(sr)](sr._column.data_array_view)
-    assert not gdf.to_string().split() == cdf.to_string().split()
+    if cudf.get_option("copy_on_write"):
+        assert gdf.to_string().split() == cdf.to_string().split()
```

Comment:
Is this not unexpected behavior? Say someone wrote this as `add_one[1, len(sr)](sr)` then I'd expect the `sr` object to be mutated with new buffers on write, which would then therefore continue to make the way the test is written be true where `cdf` is not equal to `gdf`.

Reply:
This should now be fixed.